### DISCUSSION TITLE
Custom or Ignore formatting for a specific function

### DISCUSSION THREAD

**Discussion by lucascmelo** [sql-formatter-org/sql-formatter#557] [Q&A] [answered]
Hey all.
first of all, nice work 🙌 .

So, do you guys know if I can ignore a specific formatting function on **postgresql**? 

This is my query formatted:
```
SELECT
    id,
    name
FROM
    table_abc
WHERE
    AND EXTRACT(
        MONTH
        FROM
            reporting_date
    ) = :month
    AND EXTRACT(
        YEAR
        FROM
            reporting_date
    ) = :year
```

And I would like to make the `EXTRACT` function in only one line.

```
SELECT
    id,
    name
FROM
    table_abc
WHERE
    AND EXTRACT( MONTH FROM reporting_date ) = :month
    AND EXTRACT( YEAR FROM reporting_date ) = :year
```

**Comment by nene** [comment, marked as answer]
It's really a fundamental bug/shortcoming of the formatter: it thinks it's looking at a `SELECT` statement inside the `EXTRACT()` because of the `FROM` keyword.

Unfortunately no good workaround for this.

I've started to work on [a brand new formatting tool](https://github.com/nene/prettier-plugin-sql-cst), which won't be fooled by situations like this. It's still in early stages of development though and it doesn't yet support Postgres (well, for now it only supports subset of SQLite).

**Reply by lucascmelo** [reply]
thanks for the quickly reply @nene
